### PR TITLE
Arcade-powered source-build: stage 1

### PR DESCRIPTION
**Pull request by crummel**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/857

Regression? No. Last working version: N/A

## Description

This is the first part of [arcade-powered source-build](https://github.com/dotnet/source-build/blob/release/5.0/Documentation/planning/arcade-powered-source-build/onboarding/local-onboarding.md) changes.  The goal is to enable each repo to be able to build source-build-clean on its own, and be able to aggregate this into a product that is shippable in open-source Linux distributions.

This PR adds the local build infrastructure that lets ArPow (arcade-powered source-build) run in this repo. See <https://github.com/dotnet/source-build/blob/master/Documentation/planning/arcade-powered-source-build/onboarding/local-onboarding.md> for more details about how it works.

To try it out locally, run this on Linux: `./build.sh -c Release --restore --build --pack /p:ArcadeBuildFromSource=true -bl`

This PR should have no effect on ordinary builds, or CI. ArPow stage 2 will add source-build to CI: PR validation and official builds.

For <https://github.com/dotnet/source-build/blob/master/Documentation/planning/arcade-powered-source-build/implementation-plan.md>.

This PR is effectively blocked on https://github.com/NuGet/Home/issues/10646 as we can't test it out without that change.
## PR Checklist

- [ ] PR has a meaningful title
- [ ] PR has a linked issue.
- [ ] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

File: eng/SourceBuild.props [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<Project>
+
+  <PropertyGroup>
+    <GitHubRepositoryName>nuget-client</GitHubRepositoryName>
+    <SourceBuildManagedOnly>true</SourceBuildManagedOnly>
+  </PropertyGroup>
+
+  <Target Name="ApplySourceBuildPatchFiles"
```
Comment (dominoFire) [inline eng/SourceBuild.props:8]: I removed some patches for source-build 5.0, 

https://github.com/NuGet/NuGet.Client/pull/3511

Do I need to remove those patches from dotnet/source-build@release/5.0 branch ?

**Review by dominoFire — COMMENTED** [review]
Thanks for your contribution, @crummel !

I'm working on having a green build for this PR. I left one question.

File: eng/SourceBuild.props [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<Project>
+
+  <PropertyGroup>
+    <GitHubRepositoryName>nuget-client</GitHubRepositoryName>
```
Comment (dominoFire) [inline eng/SourceBuild.props:4]: Shoud it be  NuGet.Client as the repo name?

```suggestion
    <GitHubRepositoryName>NuGet.Client</GitHubRepositoryName>
```

https://github.com/dotnet/source-build/blob/master/Documentation/planning/arcade-powered-source-build/onboarding/local-onboarding.md#engsourcebuildprops

File: eng/SourceBuild.props (hunk 1 above)
Comment (crummel) [reply eng/SourceBuild.props:8]: We will take care of that when we uptake these changes.  Thanks!

File: eng/SourceBuild.props (hunk 2 above)
Comment (crummel) [reply eng/SourceBuild.props:4]: Yes, thanks.

**Review by dominoFire — APPROVED** [review]

**Comment by nkolev92** [comment]
@dominoFire 🔔

**Comment by MichaelSimons** [comment]
Is this superseded by https://github.com/NuGet/NuGet.Client/pull/4105?  If so can it be closed?

**Comment by crummel** [comment]
Where you still planning on merging this @dominoFire?  I'm still building off it, or I can roll it into the other PR.

**Comment by crummel** [comment]
Closed, redundant with #4105.

**Pull request closed without merging**